src/main/java/com/coursesPlatform/trainer/TrainerService.java

```diff
@@ -28,30 +28,23 @@ public TrainerService(TrainerRepository trainerRepository) {
                 .collect(Collectors.toList());
     }
 
-    List<TrainerDTO> showAllTrainers() {
-        return trainerRepository.findAll()
-                .stream()
-                .map(this::convertTrainerToInternalDTO)
-                .collect(Collectors.toList());
-    }
-
     TrainerDTO add(TrainerDTO trainerDTO) {
-
-        Optional<Trainer> titleAndDescription = trainerRepository
+        Optional<Trainer> byNameAndLastName = trainerRepository
                 .findByNameAndLastName(trainerDTO.getName(), trainerDTO.getLastName());
-
-
+        if ( byNameAndLastName.isPresent() ) {
+            throw new IllegalStateException("Trainer with given name and lastname already exists");
+        }
         Trainer trainer = trainerRepository.save(convertDTOToTrainer(trainerDTO));
         return convertTrainerToDTO(trainer);
     }
 
-    TrainerDTO update(TrainerDTO updatedTrainer) throws TrainerNotFoundException {
+    TrainerDTO update(String firstname, String lastname, TrainerDTO updatedTrainer) throws TrainerNotFoundException {
```

Comment:
unchecked exceptions (line RuntimeException) doesn't have to be declared in method definition